backend/common/connect.py

```
@@ -25,7 +29,46 @@ def get_token(db: Database) -> dict | None:
+    """Proactively refresh token if it's expired or expiring soon. Returns True if refresh attempted."""
+    import logging
+    logger = logging.getLogger(__name__)
+    
+    token = get_token(db)
+    session_id = get_session_id()
+    
+    if is_token_expired_or_expiring_soon(token):
+        logger.info(f"Token expired or expiring soon for session {session_id}, attempting proactive refresh")
+        try:
+            # Create OAuth session and let it handle the refresh
+            oauth_session = get_oauth_session(db)
```

Ruff (0.12.2)

Local variable `oauth_session` is assigned to but never used

Remove assignment to unused variable `oauth_session`

(F841)